src/itest/java/uk/gov/companieshouse/company/links/steps/CompanyLinksSteps.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.companieshouse.company.links.steps;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.commons.io.FileUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.util.ResourceUtils;
+import uk.gov.companieshouse.company.links.service.CompanyProfileService;
+import uk.gov.companieshouse.stream.EventRecord;
+import uk.gov.companieshouse.stream.ResourceChangedData;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CompanyLinksSteps {
+
+    private static WireMockServer wireMockServer;
+
+    @Autowired
+    private CompanyProfileService companyProfileService;
+
+    @Autowired
+    public KafkaTemplate<String, ResourceChangedData> kafkaTemplate;
+
+    @Given("Company links consumer api service is running")
+    public void company_links_consumer_api_service_is_running() {
+        wireMockServer = new WireMockServer(8888);
+        wireMockServer.start();
+        configureFor("localhost", 8888);
+
+        stubCompanyProfileServiceCalls();
+
+        assertThat(companyProfileService).isNotNull();
+    }
+
+    @When("a message is published to the topic {string}")
+    public void a_message_is_published_to_the_topic(String topicName) throws InterruptedException {
+        kafkaTemplate.send(topicName, createMessage());
+
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        countDownLatch.await(5, TimeUnit.SECONDS);
+    }
+
+    @Then("the insolvency consumer should consume and process the message")
+    public void the_insolvency_consumer_should_consume_and_process_the_message() {
+        verify(1, getRequestedFor(urlPathEqualTo("/company/00006400")));
+        verify(0, patchRequestedFor(urlPathEqualTo("/company/00006400/links")));
```

Comment:
Shouldn't we assert consumer has added links section and sending that in request body? GET will return a profile without insolvency links, and PATCH verifies we added updated at/by and links etc

Reply:
Please let's have a chat on Monday. I think we need to define a couple of scenarios here.
Example:
Given Company Profile API service is running
When a message is published to the topic "stream-insolvency"
Then the Company Links Consumer send a GET request to the Company Profile API
And if the Links are not present 
Then the Company Links Consumer send a PATCH request to the Company Profile API

Reply:
makes sense to have scenario for get and put requests. Will add it as a new scenario